stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java

```diff
@@ -314,10 +315,27 @@ public static void prepareOutboundResponse(Context context, HTTPCarbonMessage in
 
         HttpUtil.checkEntityAvailability(context, outboundResponseStruct);
         HttpUtil.setKeepAliveHeader(context, outboundResponseMsg);
+        setCompressionHeader(context, outboundResponseMsg);
```

Comment:
Service annotation is accessed within setKeepAliveHeader method too. Better if you could move setKeepAliveHeader and setCompressionHeader to a common function.